src/main/java/be/syntra/devshop/DevshopBack/services/ProductServiceImpl.java

```diff
@@ -58,17 +58,114 @@ public Product findById(Long id) {
     }
 
     @Override
-    public List<Product> findAllByArchivedFalse() {
-        return productRepository.findAllByArchivedFalse();
+    public Page<Product> findAllByArchivedFalse(Pageable pageable) {
+        return productRepository.findAllByArchivedFalse(pageable);
+    }
+
+    @Override
+    public Page<Product> findMaxPriceProductByArchivedFalse() {
+        return productRepository.findAllByArchivedFalse(PageRequest.of(0,1, Sort.by("price").descending()));
```

Comment:
extract 
private PageRequest defaultSortDesc() {
        return PageRequest.of(0,1, Sort.by("price").descending());
    }
same for defaultSortAsc? 
As agreed, we keep these JPA-queries and (hopefully), we'll be able to change them to 1 SQL or JPQL query afterwards

Reply:
extracted both
1 query will be the goal :-)